clustermesh/clustermesh.go

```diff
@@ -1885,3 +1941,199 @@ func DisableWithHelm(ctx context.Context, k8sClient *k8s.Client, params Paramete
+		k.Log("❌ Unable to find Helm release for the remote cluster")
+		return err
+	}
+	remoteHelmValues := remoteRelease.Config
+	// Expand those values to include the clustermesh configuration
+	remoteHelmValues, err = updateClustermeshConfig(remoteHelmValues, aiRemote, aiLocal)
+	if err != nil {
+		return err
+	}
+
+	upgradeParams := helm.UpgradeParameters{
+		Namespace:   k.params.Namespace,
+		Name:        defaults.HelmReleaseName,
+		Values:      localHelmValues,
+		ResetValues: false,
+		ReuseValues: true,
+	}
+
+	// Enable clustermesh using a Helm Upgrade command against our target cluster
+	k.Log("⚠️ Configuring Cilium in cluster '%s' to connect to cluster '%s'",
+		localClient.ClusterName(), remoteClient.ClusterName())
+	_, err = helm.Upgrade(ctx, localClient.RESTClientGetter, upgradeParams)
+	if err != nil {
+		return err
+	}
+
+	// Enable clustermesh using a Helm Upgrade command against the remote cluster
+	k.Log("⚠️ Configuring Cilium in cluster '%s' to connect to cluster '%s'",
+		remoteClient.ClusterName(), localClient.ClusterName())
+	upgradeParams.Values = remoteHelmValues
+	_, err = helm.Upgrade(ctx, remoteClient.RESTClientGetter, upgradeParams)
+	if err != nil {
+		return err
+	}
+
+	k.Log("✅ Connected cluster %s and %s!", localClient.ClusterName(), remoteClient.ClusterName())
+	return nil
+}
+
+func updateClustermeshConfig(
+	values map[string]interface{},
+	aiLocal, aiRemote *accessInformation,
+) (map[string]interface{}, error) {
+	// get current clusters config slice, if it exists
+	c, found, err := unstructured.NestedFieldCopy(values, "clustermesh", "config", "clusters")
+	if err != nil || !found {
+		// allocate an empty slice, since there isn't one
+		c = []map[string]interface{}{}
+	}
+	oldClusters, ok := c.([]map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("existing clustermesh.config.cluters array is invalid")
+	}
+
+	// allocate new cluster entries
+	newClusters := []map[string]interface{}{
+		{
+			"name": aiLocal.ClusterName,
+			"ips":  []string{aiLocal.ServiceIPs[0]},
+			"port": aiLocal.ServicePort,
+		},
+		{
+			"name": aiRemote.ClusterName,
+			"ips":  []string{aiRemote.ServiceIPs[0]},
+			"port": aiRemote.ServicePort,
+		},
+	}
+
+	// merge new clusters on top of old clusters
+	clusters := map[string]map[string]interface{}{}
+	for _, c := range oldClusters {
+		name, ok := c["name"].(string)
+		if !ok {
+			return nil, fmt.Errorf("existing clustermesh.config.clusters array is invalid")
+		}
+		clusters[name] = c
+	}
+	for _, c := range newClusters {
+		clusters[c["name"].(string)] = c
+	}
+
+	outputClusters := make([]map[string]interface{}, 0)
+	for _, v := range clusters {
+		outputClusters = append(outputClusters, v)
+	}
+
+	// merge in other required values
+	baseValues := map[string]interface{}{
+		"clustermesh": map[string]interface{}{
+			"config": map[string]interface{}{
+				"enabled":  true,
+				"clusters": outputClusters,
+			},
+		},
+	}
+	newValues := helm.MergeMaps(values, baseValues)
```

Comment:
Given that `ReuseValues` is set, it shouldn't be necessary to manually merge back the full original values here.